gnomad_qc/v4/annotations/vrs_annotation_batch_v4.py

```diff
@@ -0,0 +1,523 @@
+    }
+    retry curl_and_python
+    """
+    )
+    if mount:
+        job.cloudfuse(mount, "/local-vrs-mount")
+    return job
+
+
+def main(args):
+    """Generate VRS annotations for a Hail Table of variants."""
+    # Initialize Hail w/chosen mode (spark or batch for QoB), setting global
+    # seed for if user decides to downsample
+    hl.init(
+        backend=args.backend_mode,
+        tmp_dir=args.tmp_dir_hail,
+        gcs_requester_pays_configuration="broad-mpg-gnomad",
+        default_reference="GRCh38",
+        global_seed=args.hail_rand_seed,
+        regions=["us-central1"],
+    )
+
+    logger.info(f"Hail version as: {hl.version()}")
+
+    # Example schema
+    logger.info("Example VRS schema for 2 variants:")
+    ht_example = hl.read_table("gs://gnomad-vrs-io-finals/ht-inputs/two-snps-schema.ht")
+    ht_example.show()
+
+    working_bucket = args.working_bucket
+    version = args.version
+
+    # Prefix to create custom named versions of outputs
+    prefix = args.prefix + version
+
+    input_paths_dict = {
+        "v4.0_exomes": f"gs://gnomad-qin/v4_annotations/v4_vds_all_variants.ht",
+        "test_v4.0_exomes": f"gs://gnomad-qin/v4_annotations/v4_vds_2_partitions.ht",
+        "v4.0_supp1": f"gs://gnomad-qin/v4_annotations/v4_vds_supp1.ht",
+    }
+
+    output_paths_dict = {
+        "v4.0_exomes": v4_vrs_annotations().path,
+        "test_v4.0_exomes": (
+            f"gs://{working_bucket}/v4_annotations/v4_vds_2_partitions_output.ht",
+        ),
+        "v4.0_supp1": f"gs://{working_bucket}/v4_annotations/v4_vds_supp1_output.ht",
+    }
+
+    # Read in Hail Table, partition, and export to sharded VCF
+    ht_original = hl.read_table(input_paths_dict[version])
+    assembly = get_reference_genome(ht_original.locus).name
+
+    # Option to downsample for testing, if you want to annotate part of a Hail Table but not all of it
+    # For this, is important that we have set the Hail random seed!
+    if args.downsample < 1.00:
+        logger.info("Downsampling Table...")
+        ht_original = ht_original.sample(args.downsample)
+        ht_original = ht_original.annotate_globals(vrs_downsample=args.downsample)
+
+    if args.run_vrs:
+        if args.backend_mode == "spark":
+            raise ValueError(
+                'Annotation step --run-vrs can only be run with "batch" setting for'
+                " backend-mode"
+            )
+
+        # Create backend and batch for coming annotation batch jobs
+        backend = hb.ServiceBackend(
+            billing_project=args.billing_project,
+            bucket=working_bucket,
+            remote_tmpdir=args.tmp_dir_hail,
+        )
+
+        batch_vrs = hb.Batch(name="vrs-annotation", backend=backend)
+
+        # Check resource existence
+        check_resource_existence(
+            output_step_resources={
+                "--run-vrs": [
+                    f"gs://gnomad-vrs-io-finals/ht-outputs/annotated-checkpoint-VRS-{prefix}.ht"
```

Comment:
this path isn't used anywhere

Reply:
It's indeed irrelevant to my v4 run, should I just delete check_resource_existence?

Reply:
Now we will use `hailctl batch submit` to run it on the backend.